packages/contracts-bedrock/src/cannon/MIPS2.sol

```diff
@@ -205,7 +205,7 @@ contract MIPS2 is ISemver {
                     return onWaitComplete(state, thread, true);
                 } else {
                     uint32 mem = MIPSMemory.readMem(
-                        state.memRoot, thread.futexAddr & 0xFFffFFfc, MIPSMemory.memoryProofOffset(MEM_PROOF_OFFSET, 1)
+                        state.memRoot, thread.futexAddr, MIPSMemory.memoryProofOffset(MEM_PROOF_OFFSET, 1)
```

Comment:
Worth highlighted that I removed the silent alignment here since we're now aligning the `futexAddr` upfront.  Seems like we should only do this in one place but open to discussing.

Reply:
I think we should align `futexAddr` here to be consistent with the semantics of state transitions that read memory. Our VM model is mostly context free so there should exist a transition even for States that contain an unaligned `futexAddr`. We generally don't want to create to many invalid states in our VM model because it's harder to reason about and write tests for.
Since we don't consider unaligned `futexAddr` as invalid, it follows that we don't revert if we see one and thus we should silently align the `futexAddr` here.

Reply:
Sounds good - reverted this change and made the offchain impl consistent.